Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -319,19 +299,65 @@ lemma lift_braiding_hom {T X Y : C} (f : T ⟶ X) (g : T ⟶ Y) :
 lemma lift_braiding_inv {T X Y : C} (f : T ⟶ X) (g : T ⟶ Y) :
     lift f g ≫ (β_ Y X).inv = lift g f := by aesop
 
+instance (priority := low) [BraidedCategory C] : SymmetricCategory C where
+
+/-- `ChosenFiniteProducts` implies `BraidedCategory`.
+This is not an instance to prevent diamonds. -/
+def _root_.CategoryTheory.BraidedCategory.ofChosenFiniteProducts : BraidedCategory C where
+  braiding X Y := { hom := lift (snd _ _) (fst _ _), inv := lift (snd _ _) (fst _ _) }
+
+instance : Nonempty (BraidedCategory C) :=
+  ⟨.ofChosenFiniteProducts⟩
+
+instance : Subsingleton (BraidedCategory C) where
+  allEq
+  | ⟨e₁, a₁, b₁, c₁, d₁⟩, ⟨e₂, a₂, b₂, c₂, d₂⟩ => by
+      congr
+      ext
+      · exact (@braiding_hom_fst C _ ‹_› ⟨e₁, a₁, b₁, c₁, d₁⟩ ..).trans
+          (@braiding_hom_fst C _ ‹_› ⟨e₂, a₂, b₂, c₂, d₂⟩ ..).symm
+      · exact (@braiding_hom_snd C _ ‹_› ⟨e₁, a₁, b₁, c₁, d₁⟩ ..).trans
+          (@braiding_hom_snd C _ ‹_› ⟨e₂, a₂, b₂, c₂, d₂⟩ ..).symm
+
+instance : Subsingleton (SymmetricCategory C) where
+  allEq := by rintro ⟨_⟩ ⟨_⟩; congr; exact Subsingleton.elim _ _
+
+end BraidedCategory
+
 /--
-Construct an instance of `ChosenFiniteProducts C` given an instance of `HasFiniteProducts C`.
+Construct an instance of `ChosenFiniteProducts C` given
+a terminal object and limit cones over arbitrary pairs of objects.
 -/
-noncomputable
-def ofFiniteProducts
-    (C : Type u) [Category.{v} C] [Limits.HasFiniteProducts C] :
+abbrev ofChosenFiniteProducts
```

Comment:
I would rename this to `ofFiniteProducts` to reduce confusion with `ChosenFiniteProducts`.
```suggestion
abbrev ofFiniteProducts
```
(`ChosenFiniteProducts.ofChosenFiniteProducts` is quite nonsensical on its own, I guess!)

Reply:
I am not sure about this. I would prefer to keep the name to mirror `monoidalOfChosenFiniteProducts`. I agree the name `ChosenFiniteProducts.ofChosenFiniteProducts` is nonsensical but this should be replaced with `CartesianMonoidal.ofChosenFiniteProducts` very soon.

Reply:
Well I am replacing it with `CartesianMonoidalCategory.ofFiniteProducts` in #24390...

Reply:
I'm not convinced we should be dropping `chosen`. It is to contrast with `HasFiniteProducts` where the finite product is obtained by choice.